src/Serialization/serializers.jl

```diff
@@ -31,6 +31,11 @@ mutable struct SerializerState
   refs::Vector{UUID}
   io::IO
   key::Union{Symbol, Nothing}
+  type_attr_map::Dict{Type, Vector{Symbol}}
+end
+
+function attrs_list(s::SerializerState, T::Type)
```

Comment:
This function does not work for parameterized types. I believe this is due to `register_serialization_type` only registering the attribute list to the base part of the type, but `attrs_list` is passed the concrete type and thus finds nothing.
For some concrete example, you can have a look at the branch https://github.com/oscar-system/Oscar.jl/tree/lg/serialize-Lie-attrs-test (merge of this branch with https://github.com/oscar-system/Oscar.jl/pull/3980 plus https://github.com/oscar-system/Oscar.jl/pull/4020/files#r1721805863 plus some minor changes to make them load together) and execute the following:
```julia-repl
julia> L = lie_algebra(QQ, :A, 2);

julia> is_abelian(L)
false

julia> save("tmp.mrdi", L; with_attrs=true)
```
`L` is of type `AbstractLieAlgebra{QQFieldElem}` which has serialization of the attribute `is_abelian` enabled. However, this attribute does not show up in `tmp.mrdi`.

Reply:
I see the issue, maybe we can do something similar to encode_type, but, i'll need to think about this tomorrow

Reply:
I have found a IMO nicer solution in terms of `encode_type` (which I find nice because this is already used in the serialization process as a type identifier). Have a look in 781fa8e6ea7375d21e7894fae550ea84b38eb3bc (this needs some stupid fix for the Lie algebras to work that is included in the branch as well)

Reply:
this would be an alternative to your two commits from this morning

Reply:
yes, i also thought about this, but we can avoid strings, this may be the best alternative

```
julia> T = typeof(sl2)
AbstractLieAlgebra{QQFieldElem}

julia> Base.typename(T).wrapper
AbstractLieAlgebra
```

Reply:
that's an internal and undocumented implementation detail of julia Base. I am not convinced that it's a good idea to rely on that

Reply:
But isn't this exactly what we are using `encode_type` currently for? I don't understand why we should use two different methods to do basically the same thing (convert a type to an identifier)

Reply:
ok, we agree, we'll use encode_type